src/main/java/xyz/nucleoid/creator_tools/workspace/editor/NetworkedWorkspaceEditor.java

```diff
@@ -0,0 +1,132 @@
+package xyz.nucleoid.creator_tools.workspace.editor;
+
+import java.util.stream.Collectors;
+
+import net.fabricmc.fabric.api.networking.v1.PacketByteBufs;
+import net.fabricmc.fabric.api.networking.v1.ServerPlayNetworking;
+import net.minecraft.nbt.NbtCompound;
+import net.minecraft.network.PacketByteBuf;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.util.Identifier;
+import xyz.nucleoid.creator_tools.workspace.MapWorkspace;
+import xyz.nucleoid.creator_tools.workspace.WorkspaceRegion;
+import xyz.nucleoid.map_templates.BlockBounds;
+
+/**
+ * An editor implementation that uses {@linkplain WorkspaceNetworking networking} for use with clientside mods.
+ */
+public class NetworkedWorkspaceEditor implements WorkspaceEditor {
+    private final ServerPlayerEntity player;
+    private final MapWorkspace workspace;
+
+    public NetworkedWorkspaceEditor(ServerPlayerEntity player, MapWorkspace workspace) {
+        this.player = player;
+        this.workspace = workspace;
+    }
+
+    @Override
+    public void onEnter() {
+        if (this.canSendPacket(WorkspaceNetworking.WORKSPACE_ENTER_ID)) {
+            var buf = PacketByteBufs.create();
+
+            buf.writeIdentifier(this.workspace.getIdentifier());
+            WorkspaceNetworking.writeBounds(buf, this.workspace.getBounds());
+            buf.writeIdentifier(this.workspace.getWorld().getRegistryKey().getValue());
+            buf.writeNbt(this.workspace.getData());
+
+            this.sendPacket(WorkspaceNetworking.WORKSPACE_ENTER_ID, buf);
+        }
+
+        if (this.canSendPacket(WorkspaceNetworking.WORKSPACE_REGIONS_ID)) {
+            var groups = this.workspace.getRegions().stream().collect(Collectors.groupingBy(WorkspaceRegion::marker));
+
+            for (var entry : groups.entrySet()) {
+                var buf = PacketByteBufs.create();
+
+                buf.writeString(entry.getKey());
+
+                for (var region : entry.getValue()) {
+                    buf.writeInt(region.runtimeId());
```

Comment:
Might just be unfamiliar with Yarn mappings- the spec describes these as being VarI32, but I’m not sure that’s what’s being used here? 🙂

Reply:
Fixed in 64643ced7d8e8aeb53f79221a624972cb7cfd4dd